frontend/pages/upload.vue

```diff
@@ -0,0 +1,407 @@
+<script setup lang="ts">
+import { type FileUploadUploaderEvent } from 'primevue/fileupload'
+import { serializeError } from 'serialize-error'
+import { formatFileSize } from '@/lib/filesize'
+
+const pactaClient = usePACTA()
+const { $axios } = useNuxtApp()
+const { t } = useI18n()
+
+const prefix = 'Upload'
+const tt = (key: string) => t(`${prefix}.${key}`)
+
+enum FileStatus {
+  Selected = 'Selected',
+  Waiting = 'Waiting',
+  Uploading = 'Uploading',
+  Uploaded = 'Uploaded',
+  Validating = 'Validating',
+  CleanUp = 'Cleaning Up',
+  Done = 'Done',
+  Error = 'Error',
+}
+
+interface FileState {
+  file: File
+  status: FileStatus
+  shortName: string
+  incompleteUploadId?: string
+  errorMessage?: string
+}
+
+interface FileStateDetail extends FileState {
+  index: number
+  icon: string
+  sizeStr: string
+  key: string
+  effectiveError?: string | undefined
+}
+
+const holdingsDate = useState<Date>(`${prefix}.holdingsDate`, () => new Date())
+const errorCode = useState<string>(`${prefix}.errorCode`, () => '')
+const errorMessage = useState<string>(`${prefix}.errorMessage`, () => '')
+const startedProcessing = useState<boolean>(`${prefix}.startedProcessing`, () => false)
+const isProcessing = useState<boolean>(`${prefix}.isProcessing`, () => false)
+const fileStates = useState<FileState[]>(`${prefix}.fileState`, () => [])
+
+const reset = () => {
+  holdingsDate.value = new Date()
+  errorCode.value = ''
+  errorMessage.value = ''
+  startedProcessing.value = false
+  isProcessing.value = false
+  fileStates.value = []
+}
+
+const removeFile = (index: number) => {
+  fileStates.value.splice(index, 1)
+}
+const hasAnyState = (status: FileStatus): boolean => {
+  return fileStates.value.some((fileState) => fileState.status === status)
+}
+const hasAllState = (status: FileStatus): boolean => {
+  return fileStates.value.every((fileState) => fileState.status === status)
+}
+const setAllStates = (status: FileStatus) => {
+  fileStates.value.forEach((fileState) => {
+    fileState.status = status
+  })
+}
+
+const fileStatesWithDetail = computed<FileStateDetail[]>(() => {
+  const dupeKey = (fileState: FileState) => `${fileState.file.name}-${fileState.file.size}`
+  const fileNameAndSizeCounts = fileStates.value.reduce((acc, fileState) => {
+    const key = dupeKey(fileState)
+    acc.set(key, (acc.get(key) ?? 0) + 1)
+    return acc
+  }, new Map<string, number>())
+  const isDuplicate = (fileState: FileState) => (fileNameAndSizeCounts.get(dupeKey(fileState)) ?? 0) > 1
+  return fileStates.value.map((fileState, index) => {
+    let icon = 'pi pi-spinner pi-spin'
+    if (fileState.status === FileStatus.Done) {
+      icon = 'pi pi-check-circle text-success'
+    } else if (fileState.status === FileStatus.Error) {
+      icon = 'pi pi-times text-error'
+    } else if (fileState.status === FileStatus.Selected) {
+      icon = 'pi pi-circle'
+    }
+    let otherError: string | undefined
+    // TODO(grady) validate this server side too.
```

Comment:
I filed #79 if you want to reference that here, because I wanted to add some context for how we have to do this when working with Azure.